app.jsx

```diff
@@ -11,69 +11,75 @@ import styled from "styled-components";
+    if (isItalic) {
+        style = [...style, BASE_STYLES.ITALIC];
+    }
+    if (isBold) {
+        style = [...style, BASE_STYLES.BOLD];
+    }
+    
+    return style.length > 0 ? style : [];
+}
+
+const getStyle = ({name, attribs}) => {
+    const tagStyles = (name && TAG_STYLES[name] && [TAG_STYLES[name]]) || [];
+    const inlineStyles = (attribs && attribs.style && getInlineStyle(attribs.style)) || [];
+    const aggregatedStyles = [...tagStyles, ...inlineStyles];
+    return aggregatedStyles.length > 0 ? aggregatedStyles : [BASE_STYLES.NORMAL];
+}
+
+const parseNodes = (nodes, el) => {
     let parsed = [];
     for (const node of nodes) {
-        const { attribs, children, data, name } = node;
-        if (!name) {
+        const { children, data, parent } = node;
+
+        let elementStyle;
+        if (!el) {
+            elementStyle = new Set(getStyle(parent));
+        } else {
+            getStyle(parent).forEach(style => el.add(style));
+            elementStyle = el;
+        } 
+
+        if (data) {
+            if (elementStyle.has(BASE_STYLES.BOLD) || elementStyle.has(BASE_STYLES.ITALIC)) {
+                elementStyle.delete(BASE_STYLES.NORMAL);
+            };
+
             parsed = parsed.concat({
-                style: baseStyle,
+                style: Array.from(elementStyle),
                 content: data,
             });
-        } else if (name === "b") {
-            parsed = parsed.concat(
-                parseNodes(
-                    children,
-                    baseStyle === "italic" ? "bold-italic" : "bold"
-                )
-            );
-        } else if (name === "i") {
-            parsed = parsed.concat(
-                parseNodes(
-                    children,
-                    baseStyle === "bold" ? "bold-italic" : "italic"
-                )
-            );
-        } else if (name === "span") {
-            const { style } = attribs;
-            // The detection of attributes here might be too specific. Is this
-            // really the best way to do this?
-            const isItalic = !!style.match(/italic/);
-            const isBold = !!style.match(/weight:600/);
-            if (isItalic && !isBold) {
-                parsed = parsed.concat(parseNodes(children, "italic"));
-            } else if (!isItalic && isBold) {
-                parsed = parsed.concat(parseNodes(children, "bold"));
-            } else if (isItalic && isBold) {
-                parsed = parsed.concat(parseNodes(children, "bold-italic"));
-            } else {
-                parsed = parsed.concat(parseNodes(children, "normal"));
-            }
-        }
+        } else {
+            parsed = parsed.concat(parseNodes(children, elementStyle));
+        };
     }
     return parsed;
-};
+}
 
 const parseHtml = (html) =>
     ReactHtmlParser(html, {
         transform: (node, i) => {
-            const { children, name, parent } = node;
-            if (!parent && name === "div") {
```

Comment:
1. I started with investigating why when I pasted code, most of the time I would see an empty json array returned. I realised it was because of the name === "div", which didn't apply to most things I was pasting, so I removed it.